vms/platformvm/state/state.go

```diff
@@ -1757,6 +1764,165 @@ func (s *state) write(updateValidators bool, height uint64) error {
 	)
 }
 
+func (s *state) writeWeightDiffs(height uint64, weightDiffs map[subnetNodeKey]*ValidatorWeightDiff) error {
```

Comment:
down the line (https://github.com/ava-labs/avalanchego/pull/2086) we may consider making this a free function and explicitly pass the dbs to write data to among parameters.
Just to highlight that these write helpers should not access anything related to state but dbs